### PR TITLE
Dev/144895 create new endpoint to provide product updates since datetime

### PR DESCRIPTION
**Pull request by pratibh14837**
This PR contains coding, unit test cases in WIP.
Please review the same

Thanks

### CODE REVIEW

**Review by richardAhz — CHANGES_REQUESTED** [review]
Can we change  
GetExchangeSetDataProductDateTimeAsync
to
GetExchangeSetProductDataByDateTimeAsync

**Review by richardAhz — APPROVED** [review]
LGTM

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/UKHO.ExchangeSetService.API.UnitTests.csproj [hunk 1]
```diff
@@ -28,10 +28,6 @@
     </PackageReference>
   </ItemGroup>
 
-  <ItemGroup>
-    <Folder Include="Controllers\" />
-  </ItemGroup>
```
Comment (gaurav15233) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/UKHO.ExchangeSetService.API.UnitTests.csproj:33]: Please remove this deleted line. This should not be in commit

**Review by gaurav15233 — COMMENTED** [review]
Suggested Comments

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductInformationController.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        /// <br/><para><i>Example</i> : Wed, 21 Oct 2015 07:28:00 GMT</para>
+        /// </param>
+        /// <response code="200">A JSON body that containing the information of ENC versions that have been issued since that datetime.</response>
+        /// <response code="304">Not modified.</response>
+        /// <response code="400">Bad Request.</response>
+        /// <response code="401">Unauthorised - either you have not provided any credentials, or your credentials are not recognised.</response>
+        /// <response code="403">Forbidden - you have been authorised, but you are not allowed to access this resource.</response>
+        /// <response code="429">You have sent too many requests in a given amount of time. Please back-off for the time in the Retry-After header (in seconds) and try again.</response>
+        /// <response code="500">Internal Server Error.</response>
+
```
[the remark this answers is not in the document]
Comment (pratibh14837) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductInformationController.cs]: removed blank line

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductInformationController.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+using System.Net;
+using UKHO.ExchangeSetService.API.Filters;
+using UKHO.ExchangeSetService.Common.Models.Request;
+using UKHO.ExchangeSetService.Common.Models.Response;
+
+namespace UKHO.ExchangeSetService.API.Controllers
+{
+    [ApiController]
+    [Authorize]
+    [ServiceFilter(typeof(BespokeExchangeSetAuthorizationFilterAttribute))]
```
[the remark this answers is not in the document]
Comment (pratibh14837) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductInformationController.cs]: resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductInformationControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        private IHttpContextAccessor fakeHttpContextAccessor;
+        private ILogger<ProductInformationController> fakeLogger;
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeHttpContextAccessor = A.Fake<IHttpContextAccessor>();
+            fakeLogger = A.Fake<ILogger<ProductInformationController>>();
+            A.CallTo(() => fakeHttpContextAccessor.HttpContext).Returns(new DefaultHttpContext());
+
```
[the remark this answers is not in the document]
Comment (pratibh14837) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductInformationControllerTests.cs]: removed

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs [hunk 5]
```diff
@@ -630,5 +630,6 @@ public async Task WhenValidRequest_ThenGetProductDataSinceDateTimeReturnSuccess(
         }
 
         #endregion ProductDataSinceDateTime
+
```
[the remark this answers is not in the document]
Comment (pratibh14837) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs]: removed

**Review by jagannathsankpal — APPROVED** [review]
LGTM

**Pull request merged**